src/libstd/fs.rs

```diff
@@ -674,6 +716,41 @@ impl DirEntry {
     /// The exact text, of course, depends on what files you have in `.`.
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn path(&self) -> PathBuf { self.0.path() }
+
+    /// Return the metadata for the file that this entry points at.
+    ///
+    /// This function may not require an extra system call on some platforms
```

Comment:
How about: "This function may require an extra system call on some platforms" (or, better yet: we could go ahead and add the `# Platform behavior` section and give a fuller breakdown).